test/parse/test_entity_declaration.rb

```diff
@@ -18,25 +22,22 @@ def parse(internal_subset)
       REXML::Document.new(xml(internal_subset)).doctype
     end
 
+    public
     def test_empty
-      exception = assert_raise(REXML::ParseException) do
+      assert_raise(REXML::ParseException) do
         parse(<<-INTERNAL_SUBSET)
 <!ENTITY>
         INTERNAL_SUBSET
       end
-      assert_equal(<<-DETAIL.chomp, exception.to_s)
-Malformed notation declaration: name is missing
-Line: 5
-Position: 72
-Last 80 unconsumed characters:
- <!ENTITY>  ]> <r/>
-      DETAIL
```

Comment:
This test raises following exception:

```
<"#<NoMethodError: undefined method `captures' for nil>\n" +
"/home/watson/prj/rexml/lib/rexml/parsers/baseparser.rb:321:in `pull_event'\n" +
"/home/watson/prj/rexml/lib/rexml/parsers/baseparser.rb:225:in `pull'\n" +
"/home/watson/prj/rexml/lib/rexml/parsers/treeparser.rb:22:in `parse'\n" +
"/home/watson/prj/rexml/lib/rexml/document.rb:448:in `build'\n" +
```

I removed because I didn't think the need to check `NoMethodError` message.

Reply:
We should raise `ParseException`.
Could you fix the implementation in this PR?

Reply:
Ah, I was wrong above comment.  sorry.

It raises `ParseException`, but it has  `#<NoMethodError: undefined method `captures' for nil>` exception message.

```
#<REXML::ParseException:"#<NoMethodError: undefined method `captures' for nil>
```

Reply:
OK. Then, could you fix the `NoMethodError` error message in this PR?

Reply:
I fixed with https://github.com/ruby/rexml/pull/182/commits/93ec28f68b6c04f6c31bc8e117b586ef1a469331